src/ovp_pipeline/context_binder.py

```
@@ -0,0 +1,679 @@
+        return RetrievalContext(query=query)
+
+    try:
+        from ovp_pipeline.discovery import discover_related
+
+        rows = discover_related(
+            resolve_vault_dir(vault_dir),
+            query,
+            engine="knowledge",
+            limit=top_k,
+        )
+    except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)